prepare.go

```diff
@@ -595,3 +600,76 @@ func (entry *SubmissionQueueEntry) PrepareCmdSock(
 	entry.prepareRW(OpUringCmd, fd, 0, 0, 0)
 	entry.Off = uint64(cmdOp << bit32Offset)
 }
+
+// convert syscall.Sockaddr interface to raw pointers
+// ref: https://groups.google.com/g/golang-nuts/c/B-meiFfkmH0/m/PyFC84kPaFkJ
+func sockaddr(addr syscall.Sockaddr) (unsafe.Pointer, uint32, syscall.Errno) {
```

Comment:
There is an easier way to convert syscall.Sockaddr to a pointer. We can link private function from golang standard library as follows:

//go:linkname sockaddr syscall.Sockaddr.sockaddr
func sockaddr(addr syscall.Sockaddr) (unsafe.Pointer, uint32, error)